### DISCUSSION TITLE
Redirecting to a new route should re-render page

### DISCUSSION THREAD

**Discussion by colynb** [remix-run/remix#4436] [Q&A]
I have a site that redirects to another route to reset a cookie then redirects back to refresh the page using new cookie data. A similar example would be switching to another language then redirecting back and re-rendering the page. I cannot get the page to re-render unless I do the refresh myself. I'm not sure what I am doing wrong.

**Comment by kiliman** [comment]
Hmm.. cookie changes should force revalidation. Can you post some code?

**Reply by colynb** [reply]
Thank you! I’ll have to come up with a reproducible small example. I’ll let you know in the next day or two.
